### PR TITLE
[EH] Checkpoint

### PR DESCRIPTION
**Pull request by l0lawrence**

### CODE REVIEW

File: sdk/eventhub/azure-eventhub/azure/eventhub/aio/_consumer_async.py [hunk 1]
```diff
@@ -136,9 +136,21 @@ def __init__(self, client: "EventHubConsumerClient", source: str, **kwargs: Any)
         self._last_callback_called_time = None
         self._callback_task_run = None
 
-    def _create_handler(
+    async def _create_handler(
         self, auth: Union["uamqp_JWTTokenAsync", JWTTokenAuthAsync]
     ) -> None:
+        # TODO: what happens if consumer has event position set and checkpoint store is used?
+        if self._client._checkpoint_store: # this is needed if no checkpoint store will fail
```
Comment (l0lawrence) [inline sdk/eventhub/azure-eventhub/azure/eventhub/aio/_consumer_async.py:143]: with in_memory checkpoint this fails, so we need this `if'

Comment (l0lawrence) [reply sdk/eventhub/azure-eventhub/azure/eventhub/aio/_consumer_async.py:143]: probably need to investigate that more bc in_memory also gets assigned to checkpoint-store? maybe just not on the ._client level but in the event processor itself

Comment (l0lawrence) [reply sdk/eventhub/azure-eventhub/azure/eventhub/aio/_consumer_async.py:143]: is there a self._checkpoint_store option?

Comment (swathipil) [reply sdk/eventhub/azure-eventhub/azure/eventhub/aio/_consumer_async.py:143]: I think we could set self._checkpoint_store in the constructor if we wanted to. Any reason for not calling the one on the client?

**Pull request closed without merging**